### DISCUSSION TITLE
modelID: 'TS0601', manufacturerName: '_TZE200_gd4rvykv' wrongly recognized as TS0601_thermostat

### DISCUSSION THREAD

**Discussion by pioniziolek** [Koenkk/zigbee2mqtt#16642] [Devices]
Recently I added new thermostat _TZE200_gd4rvykv // Sanico to zigbee2mqtt after review it is recognized as supported device TS0601_thermostat but does not work properly. 

I'm not advanced user but have found that device looks exactly like model: 'TS0601_thermostat3. 
Please advice what to do to correct it.

**Comment by Koenkk** [comment]
Fixed!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by SnakeZi** [reply]
Fast as ever, @Koenkk  :)
I'll just add that _TZE200_gd4rvykv has the same datapoints as _TZE200_9gvruqf5

3 = work_state	(standy / heating)
8 = Window	(ON/OFF)
10 = Antifreeze (ON/OFF)
27 = Temperature Calibration (-1/0/1)
40 = Lock (ON/OFF)
101 = Switch
102 = Current	(current temperature in Celsius)
103 = Setting Temperature
104 = Heating Time
105 = Battery Level Alarm
106 = Away (same as ECO Mode) (ON/OFF)
107 = programming mode
108 = programming switch
109 = programming data (obsolate)
110 = Historical data protocol (day - set temperaturę)
111 = System Time Synchronization
112 = Historical data protocol (week - set temperature)
113 = Historical data protocol (month - set temperature)
114 = Historical data agreement (year - set temperature)
115 = Historical data protocol (day - current temperature)
116 = Historical data protocol (week - current temperature)
117 = Historical data protocol (month - current temperature)
118 = Historical data protocol (year - current temperature)
119 = Historical data protocol (day - motor opening)
120 = Historical data protocol (week – motor opening)
121 = Historical data protocol (month – motor opening)
122 = Historical data protocol (year – motor opening)
123 = Programming Data (Monday)
124  = Programming Data (Tuesday)
125 = Programming Data (Wednesday)
126  = Programming Data (Thursday)
127 =  Programming Data (Friday)
128 = Programming Data (Saturday)
129 = Programming Data (Sunday)
130 = Water scale proof (ON/OFF)

**Reply by Koenkk** [reply]
Added the `_TZE200_gd4rvykv`

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by SnakeZi** [reply]
As I use a stable version of Zigbee2MQTT addon in HA, I was patiently waiting for the next update. After updating to 1.30.2-1 I must say it is better.. but still far from perfect.
_TZE200_gd4rvykv is recognized as TS0601_thermostat_3 but it doesn't really work.

Below is a state shown by zigbee2mqtt
{
    "child_lock": "UNLOCK",
    "current_heating_setpoint": 19,
    "frost_protection": "ON",
    "linkquality": 14,
    "local_temperature_calibration": 0,
    "scale_protection": "ON",
    "system_mode": "off",
    "battery_low": null,
    "error": null,
    "running_state": null
}

but it doesn't reflect the real state of device where heating setpoint is set to 16* and system is in auto mode (it remembers schedules from Tuya App).
Zigbee2Mqtt also can't read the current temperature in room.


This is what I get in debug logs if I manually change heating setpoint on the device (from 15*C to 17*C).
Debug 2023-03-03 00:23:17Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,150],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:23:18Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,160],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:23:19Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,170],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:23:26Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,170],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0


And when I turn the valve to off
Debug 2023-03-03 00:25:39Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,130],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:25:39Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,90],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:25:40Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,50],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:25:40Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0],"type":"Buffer"},"datatype":1,"dp":101}],"seq":256}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:25:41Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,50],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:25:41Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,50],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0


The device completly ignores commands from zigbee2mqtt.
Debug 2023-03-03 00:29:47Received MQTT message on 'zigbee2mqtt/0xa4c1381605fccdb2/set' with data '{"current_heating_setpoint":21}'
Debug 2023-03-03 00:29:47Publishing 'set' 'current_heating_setpoint' to '0xa4c1381605fccdb2'
Info 2023-03-03 00:29:47MQTT publish: topic 'zigbee2mqtt/0xa4c1381605fccdb2', payload '{"battery_low":null,"child_lock":"UNLOCK","current_heating_setpoint":21,"error":null,"frost_protection":"ON","linkquality":40,"local_temperature_calibration":0,"running_state":null,"scale_protection":"ON","system_mode":"heat"}'
Debug 2023-03-03 00:29:50Received MQTT message on 'zigbee2mqtt/0xa4c1381605fccdb2/set' with data '{"system_mode":"auto"}'
Debug 2023-03-03 00:29:50Publishing 'set' 'system_mode' to '0xa4c1381605fccdb2'
Info 2023-03-03 00:29:50MQTT publish: topic 'zigbee2mqtt/0xa4c1381605fccdb2', payload '{"battery_low":null,"child_lock":"UNLOCK","current_heating_setpoint":21,"error":null,"frost_protection":"ON","linkquality":40,"local_temperature_calibration":0,"running_state":null,"scale_protection":"ON","system_mode":"auto"}'



How can I help to make it work?





PS. This is what I got when I click "reconfigure"
Debug 2023-03-03 00:34:02Received MQTT message on 'zigbee2mqtt/bridge/request/device/configure' with data '{"id":"0xa4c1381605fccdb2","transaction":"3kkl0-1"}'
Info 2023-03-03 00:34:02Configuring '0xa4c1381605fccdb2'
Debug 2023-03-03 00:34:06Received Zigbee message from '0xa4c1381605fccdb2', type 'readResponse', cluster 'genBasic', data '{"65534":0,"appVersion":67,"manufacturerName":"_TZE200_gd4rvykv","modelId":"TS0601","powerSource":3,"zclVersion":3}' from endpoint 1 with groupID 0
Info 2023-03-03 00:34:06Successfully configured '0xa4c1381605fccdb2'
Info 2023-03-03 00:34:07MQTT publish: topic 'zigbee2mqtt/bridge/response/device/configure', payload '{"data":{"id":"0xa4c1381605fccdb2"},"status":"ok","transaction":"3kkl0-1"}'

And after turning valve up to force it to start heating (25*C)
Debug 2023-03-03 00:36:15Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,250],"type":"Buffer"},"datatype":2,"dp":103}],"seq":768}' from endpoint 1 with groupID 0
Debug 2023-03-03 00:36:20Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[1],"type":"Buffer"},"datatype":4,"dp":3}],"seq":6912}' from endpoint 1 with groupID 0
Info 2023-03-03 00:36:20MQTT publish: topic 'zigbee2mqtt/0xa4c1381605fccdb2', payload '{"battery_low":null,"child_lock":"UNLOCK","current_heating_setpoint":21,"error":null,"frost_protection":"ON","linkquality":54,"local_temperature_calibration":0,"running_state":"idle","scale_protection":"ON","system_mode":"auto"}'

and no further messages...

until after some time
Debug 2023-03-03 00:43:17Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,196],"type":"Buffer"},"datatype":2,"dp":102}],"seq":512}' from endpoint 1 with groupID 0
..
Debug 2023-03-03 00:47:38Received Zigbee message from '0xa4c1381605fccdb2', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"dpValues":[{"data":{"data":[0,0,0,208],"type":"Buffer"},"datatype":2,"dp":102}],"seq":512}' from endpoint 1 with groupID 0
(just a guess but this could be a current temperature read by valve sensor)

**Comment by pioniziolek** [comment]
I found something strange about this thermostat - it behave exactly same via tuya integration and zigbee2mqtt while it still works OK with Tuya application.

**Comment by SnakeZi** [comment]
As this is only a discussion, I have created an issue from it: https://github.com/Koenkk/zigbee2mqtt/issues/16966
